lane-layout-gen/src/main.rs

```diff
@@ -0,0 +1,157 @@
+use std::cmp::{min, max};
+use std::collections::HashMap;
+
+use rand::{thread_rng, Rng, prelude::SliceRandom, distributions::Uniform};
+
+use clap::Parser;
+
+use kdam::par_tqdm;
+
+use rayon::prelude::*;
+
+#[derive(Parser, Debug)]
+struct Args {
+    #[arg(short)]
+    n: u64,
+
+    #[arg(short, long, default_value_t = 1)]
+    iter: isize,
```

Comment:
Make it optional and add an optional parameter: `search-to` to set a minimal target, if not get, search it forever.